tools/ld-chroma-decoder/ycbcr.h

```diff
@@ -0,0 +1,66 @@
+/************************************************************************
+
+    ycbcr.h
+
+    ld-chroma-decoder - Colourisation filter for ld-decode
+    Copyright (C) 2018 Chad Page
+    Copyright (C) 2018-2019 Simon Inns
+    Copyright (C) 2021 Phillip Blucas
+
+    This file is part of ld-decode-tools.
+
+    ld-chroma-decoder is free software: you can redistribute it and/or
+    modify it under the terms of the GNU General Public License as
+    published by the Free Software Foundation, either version 3 of the
+    License, or (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+************************************************************************/
+
+#ifndef YCBCR_H
+#define YCBCR_H
+
+#include <QCoreApplication>
+#include <QDebug>
+
+#include "yiq.h"
+
+// ITU-R BT.601-7
+#define ONE_MINUS_Kb (1 - 0.114)
```

Comment:
Don't use #define for this in C++ - make them static constexpr constants, so they're computed at compile time. Then you can specify the intended type, and you can write "static constexpr double SIN33 = sin(whatever);" rather than needing to write out the value.

Reply:
I figured there was a C++ way to do this at compile time without using the preprocessor.

Reply:
constexpr trig functions don't seem to be a thing, so I had to keep the magic values.